### ISSUE TITLE
Pinned packages in package structure

### ISSUE THREAD

**Issue by bjonen** [conda/conda#1228]
Hey guys, 

I have a package A which depends on package B. I build both myself. They both depend on various third-party libraries. 
Now I would like to be able to pin for example `pandas` to version `x`. I get the error: 
`Error Unsatisfiable package specifications` in various cases: 
1) Pin in package A to `x` and no pin in package B
2) Pin in package A to `x` and pin in package B to `x`. 

I could understand that in 1), "no pin in B" can be interpreted as newest version and therefore constitute a conflict. But 2) seems puzzling to me. 

Am I overlooking something here?

**Comment by asmeurer** [comment]
Likely one of your other dependencies has a different pandas version restriction in its dependencies. If you run the install command that gives you the unsatisfiable package specifications error with the `--alt-hint` flag, it will give you some more insight into exactly what restrictions are preventing it from working.

**Comment by asmeurer** [comment]
In your case 1, where you put no version restriction in package B, it doesn't mean "newest version". It means any version, and conda will try to pick the newest one that works (but it will try all of them).

**Comment by bjonen** [comment]
Thanks for the hints! I tracked the conflict down to the following two packages: 
1) matplotlib=1.3.1
2) seaborn
I am still a bit puzzled why this constitutes a conflict.

**Comment by bjonen** [comment]
`conda show` gives me the following information: 

```
matplotlib                   1.3.1                np17py33_0  defaults
                             1.3.1                np17py27_0  defaults
                             1.3.1                np18py33_1  defaults
                             1.3.1                np18py27_1  defaults
                             1.3.1                np17py33_1  defaults
                             1.3.1                np17py27_1  defaults
                             1.3.1                np18py34_2  defaults
                             1.3.1                np18py33_2  defaults
                          *  1.3.1                np18py27_2  defaults
                             1.3.1                np18py26_2  defaults

seaborn                      0.5.0                np19py34_0  defaults
                             0.5.0                np19py33_0  defaults
                             0.5.0                np19py27_0  defaults
                             0.5.1                np19py34_0  defaults
                             0.5.1                np19py33_0  defaults
                          *  0.5.1                np19py27_0  defaults
```
So I guess seaborn has numpy 19 as pinned dependency. Is it somehow possible to peak at the meta.yaml file?

**Comment by asmeurer** [comment]
'conda info seaborn' will show you the metadata for seaborn. But in
general, if a package has "np19" in the build string, that means it depends
on numpy 1.9.

Aaron Meurer

**Comment by bjonen** [comment]
So, the source of the problem was this: The package seaborn is not available in the continuum repos in a version that is compatible with matplotlib 1.3.1. I added a binstar repository with an older version which solved my problem. 

What could perhaps be improved is the hint. The issue here is not that the packages conflict with other packages but that seaborn is only available in 0.51 in `defaults` and version 0.2.1 is needed. 
```
c:\> conda create -n test33 python
c:\> activate test33
[test33] c:\> conda install matplotlib==1.3.1 seaborn --override-channels --channel defaults
Fetching package metadata: ....
Solving package specifications: ..
Error: Unsatisfiable package specifications.
Generating hint:
[      COMPLETE      ] |#################################################| 100%


Hint: the following combinations of packages create a conflict with the
remaining packages:
  - seaborn
  - matplotlib ==1.3.1
```
Otherwise this issue is ok for me to be closed. Thanks for the help.

**Comment by asmeurer** [comment]
Generating a good hint is a hard problem, but if you have any suggestions on how to do it better let me know.
